rules/ask-for-help/rule.md

```diff
@@ -4,14 +4,33 @@ title: Do you know when to ask for help?
 related:
   - do-you-use-timeboxing-to-avoid-wasted-time
+  - do-you-manage-up
+  - do-you-add-context-reasoning-to-your-emails
 created: 2022-08-30T04:06:01.553Z
 guid: 113b1e11-6df4-4901-9e9f-bcc06b6fb352
 ---
 When you're working, you will encounter issues that block you from progressing, force you to do a lot of research or otherwise spend time building towards being able to complete your work. In these moments, the question arises, "When should I ask for help?"
 
+Before asking for help, make sure you've done your due-diligence and exhausted your normal avenues for unblocking yourself. Here are some examples for developers:
+
+- Try Googling it
+- Check Stack Overflow
+- Read the documentation e.g. [Microsoft Docs](https://aus01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fdocs.microsoft.com%2Fen-au%2F&data=05%7C01%7CSamWagner%40ssw.com.au%7C2b80dfbf1b4c4397f32808da8aecff12%7Cac2f7c34b93548e9abdc11e5d4fcb2b0%7C0%7C0%7C637975047735355922%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=EMhkvkYBn%2BCbTaN7jP%2F9kem9MM%2FkaOPSKLHe3Gej6AM%3D&reserved=0)
+- Explore the code e.g.
+- look at if any other pages do similar things
+- put breakpoints and check the values to see if you can figure out little bits
+- Systematically check things, check a tiny part, confirm your understanding, then move to the next part
+- Try to resolve it yourself a few times (at least 30 mins)
+
+
+
 As everyone's struggles are different and everyone's way of handling pressure is different, it's hard to provide a useful metric for precisely when to ask for help. However, here's a general rule to follow:
 
 1. Find the smallest 'significant' amount of work that your project can allocate. In Scrum, this would be the smallest effort value (Extra-Small if using shirt sizes)
 2. If you spend that amount of time trying to work out the problem, you **should** ask for help
+
+Once you've determined that you need help it is vital to [draft up context about what you have done](https://aus01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fwww.ssw.com.au%2Frules%2Fdo-you-add-context-reasoning-to-your-emails&data=05%7C01%7CSamWagner%40ssw.com.au%7C2b80dfbf1b4c4397f32808da8aecff12%7Cac2f7c34b93548e9abdc11e5d4fcb2b0%7C0%7C0%7C637975047735355922%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=rNlv9Yw%2B3JuEQLF%2BMmRgdOALxNoWw8%2FQv4n0hU5Ry2U%3D&reserved=0) and [provide options so they have some starting point](https://aus01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fwww.ssw.com.au%2Frules%2Fdo-you-manage-up&data=05%7C01%7CSamWagner%40ssw.com.au%7C2b80dfbf1b4c4397f32808da8aecff12%7Cac2f7c34b93548e9abdc11e5d4fcb2b0%7C0%7C0%7C637975047735355922%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=coB2z2m0aKvqUK7GmYXVjh28aFAoHgksg82QrpUHJB4%3D&reserved=0).
```

Comment:
change to https://www.ssw.com.au/rules/do-you-add-context-reasoning-to-your-emails
and
https://www.ssw.com.au/rules/do-you-manage-up

Reply:
Changed